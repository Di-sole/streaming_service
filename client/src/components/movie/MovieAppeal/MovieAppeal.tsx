import styles from './MovieAppeal.module.scss';
import ShapedLinkUI from '@/components/UI/links/ShapedLink/ShapedLinkUI';
import { useTranslation } from 'next-i18next';

interface MovieAppealProps {
    title: string;
}

const MovieAppeal = ({ title }: MovieAppealProps) => {
    const { t } = useTranslation('movie');
    return (
        <div className={styles.container} data-testid="div-movieAppeal">
            <h2>
                {t('watch')} «{title}» {t('onAllDevices')}
            </h2>
            <p>{t('availableForDownload')}</p>
            <ShapedLinkUI
                className={styles.container__link}
                href="https://www.ivi.ru/devices"
                shape="rectangular"
                target="_blank"
            >
                {t('connectDevices')}
            </ShapedLinkUI>
        </div>
    );
};

export default MovieAppeal;
