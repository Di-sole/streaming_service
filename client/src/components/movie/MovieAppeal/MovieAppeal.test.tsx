import { render, screen } from '@testing-library/react';
import MovieAppeal from './MovieAppeal';

jest.mock('next-i18next', () => ({
    useTranslation: () => {
        return {
            t: (str: string) => str,
            i18n: {
                changeLanguage: () => new Promise(() => {}),
            },
        };
    },
}));

describe('MOVIEAPPEAL SNAPSHOTS TESTS', () => {
    test('MovieAppeal should not be changed', () => {
        render(<MovieAppeal title="title" />);

        const div = screen.getByTestId('movieAppeal');
        expect(div).toMatchSnapshot();
    });
});
