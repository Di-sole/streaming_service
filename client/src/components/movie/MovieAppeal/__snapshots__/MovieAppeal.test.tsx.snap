// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MOVIEAPPEAL SNAPSHOTS TESTS MovieAppeal should not be changed 1`] = `
<div
  class="container"
  data-testid="movieAppeal"
>
  <h2>
    watch
     «
    title
    » 
    onAllDevices
  </h2>
  <p>
    availableForDownload
  </p>
  <a
    class="container container_rectangular container__link"
    href="https://www.ivi.ru/devices"
    target="_blank"
  >
    <div
      class="container__content"
    >
      connectDevices
    </div>
  </a>
</div>
`;
